### PR TITLE
Fixes #26560 - paused tasks notification and troubleshooting info

### PR DESCRIPTION
**Pull request by iNecas**
This path adds notifications and troubleshooting info for pauses tasks

Troubleshooting info
--------------------

A new `TroubleshootingHelpGenerator` service class has been added to
provide troubleshooting info for paused tasks. By default, it provides
generic information explaining the situation and link to troubleshooting
documentation (in case Setting[:foreman_tasks_troubleshooting_url] is
set.

The troubleshooting_url can contain macros for foreman version and
normalized task label (the label without non-alhpanumeric characters).
For example, for the following url setting:

      https://tfm.org/%{version}/tasks_troubleshooting.html#%{label}

and task 'Plugin::MyTask', the generated url would be:

      https://tfm.org/1.22/tasks_troubleshooting.html#PluginMyTask

The setting is currently not set (which means the troubleshooting link to
currently generated. We will add the setting once the page is ready in
docs.

It's also possible to extend the troubleshooting info and links on
per-action basis, by defining specific action method (see
`test/support/dummy_dynflow_action.rb` for example on how to define it.

Admin notifications
-------------------

Every time there is paused task, an admin gets a notification about
current amount of paused tasks with links to:

1. list of paused tasks
2. troubleshooting links from `TroubleshootingHelpGenerator`
  - the label for the troubleshooting url is filled with `''` in this
  case.

In case new paused task is added, the previous notification is removed
and an update notification is generated, so that the user's notification
is not overloaded with data.

Owner notification
------------------

In case the task is triggered by some user (the host-related tasks are
out of scope), the owner should the notification about the single task
being paused, with links to

1. details of the task
2. troubleshooting info for the task

How to test
-----------

1. set the `foreman_tasks_troubleshooting_url` config to some url
(as example might be
https://access.redhat.com/solutions/satellite6-tasks)
2. make some tasks to get paused
3. look at the notification drawer
4. loot the the details of the task

### CODE REVIEW

File: app/models/foreman_tasks/task.rb [hunk 1]
```diff
@@ -148,6 +148,22 @@ def self_and_parents
       end
     end
 
+    def notification_recipients_ids
```
[the remark this answers is not in the document]
Comment (iNecas) [reply app/models/foreman_tasks/task.rb]: This is a notifcations thing https://github.com/theforeman/foreman/blob/745ff85013db740b093a4c8a86e4ef3285f383f1/app/models/notification.rb#L47

**Comment by adamruzicka** [comment]
Needs a rebase

**Comment by iNecas** [comment]
Rebased: Regarding the notification on sub-action, I lean towards right now limiting us on the main action only: the troubleshooting docs can talk more about failing the specific sub-action in the text, but we're mainly focused on how to get the paused task resolved ~ the main action. I will try to fix that.

File: app/models/foreman_tasks/task.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (iNecas) [reply app/models/foreman_tasks/task.rb]: Added explanatory comment in the code

**Comment by iNecas** [comment]
@adamruzicka all the comments should be addressed

**Comment by iNecas** [comment]
Also, after offline discussion, I've removed the trimming of `::` from the label for the anchor purposes

**Comment by adamruzicka** [comment]
Still APJ

**Comment by iNecas** [comment]
Seems like test failures are related for some weird reason

File: test/foreman_tasks_test_helper.rb [hunk 2]
```diff
@@ -7,13 +7,16 @@
...
 require 'dynflow/testing'
+require 'foreman_tasks/test_helpers'
 
 FactoryBot.definition_file_paths = ["#{ForemanTasks::Engine.root}/test/factories"]
 FactoryBot.find_definitions
 
 ForemanTasks.dynflow.require!
 ForemanTasks.dynflow.config.disable_active_record_actions = true
 
+ForemanTasks::TestHelpers.use_in_memory_sqlite!
```
Comment (iNecas) [inline test/foreman_tasks_test_helper.rb]: Seems like using the same db for standard test world and the in-thread executor one was causing issues: this should make the things more isolated.

**Comment by iNecas** [comment]
The tests seems to be much happier now. Should by ready for final review

File: test/unit/actions/action_with_sub_plans_test.rb [hunk 3]
```diff
@@ -35,7 +35,7 @@ def run; end
         user = FactoryBot.create(:user)
         triggered = ForemanTasks.trigger(ParentAction, user)
         raise triggered.error if triggered.respond_to?(:error)
-        triggered.finished.wait(2)
+        triggered.finished.wait(30)
```
Comment (adamruzicka) [inline test/unit/actions/action_with_sub_plans_test.rb]: Does this need to stay increased when the root cause was resolved?

**Comment by adamruzicka** [comment]
[test foreman_tasks]

**Comment by adamruzicka** [comment]
[test foreman-tasks]

File: test/unit/actions/action_with_sub_plans_test.rb (hunk 3 above)
Comment (iNecas) [reply test/unit/actions/action_with_sub_plans_test.rb]: I would leave it there

**Comment by iNecas** [comment]
All green now

**Comment by iNecas** [comment]
@adamruzicka anything else to do here or ok to merge?

**Comment by adamruzicka** [comment]
Nope, looks good to me

**Comment by adamruzicka** [comment]
Thanks @iNecas !

**Pull request merged**